src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -170,10 +182,15 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
             ImmutableArray<BoundExpression> boundNamedArguments = analyzedArguments.NamedArguments;
             constructorArguments.Free();
 
-            return new BoundAttribute(node, attributeConstructor, boundConstructorArguments, boundConstructorArgumentNamesOpt,
+            var attribute = new BoundAttribute(node, attributeConstructor, boundConstructorArguments, boundConstructorArgumentNamesOpt, argsToParamsOpt, expanded,
                 boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
-        }
 
+            if (Compilation.LanguageVersion >= MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion())
+            {
+                NullableWalker.Analyze(Compilation, attribute, diagnostics);
+            }
```

Comment:
Consider adding an overload of `Analyze` which does this specific check. Perhaps call it `AnalyzeWhenNeeded` or a better name. This way in the future we can do even more advanced items like skipping it when you're using C# 8.0 but haven't used any nullability.